client/src/Header.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import './App.css';
+import './index.css';
+import HeaderButton from './headerButton';
+import HeaderMenu from './HeaderMenu'
+
+class Header extends React.Component {
+   constructor(props)  {
+      super(props)
+        this.state = {
+            name: 'I am so confused right now.',
+            visible: false,
+            
+       }
+       this.handleMenuChange=this.handleMenuChange.bind(this)
+       this.toggleHeader = this.toggleHeader.bind(this)
```

Comment:
Good, you bind these correctly here since you aren't using an arrow function to define your functions. This functionality you have here is pretty cool, having a new screen slide out from the side. Maybe you could add your links to your header, make it's height less than 100% of the screen, and show it across the top by default. Then there could be a button that hides the menu and shows the menu as desired. To do this, you would have to just set the visible state to true by default and then change some styling in your css, like the height rule.